exercises/codec-server/README.md

```diff
@@ -0,0 +1,119 @@
+* Enable CORS and review other deployment parameters
+* Integrate your Codec Server with the Temporal Web UI
+* Securely return decoded results in the CLI and the Web UI
+
+Make your changes to the code in the `practice` subdirectory (look for 
+`TODO` comments that will guide you to where you should make changes to 
+the code). If you need a hint or want to verify your changes, look at 
+the complete version in the `solution` subdirectory.
+
+
+## Part A: Configure a Codec Server to Use Your Data Converter
+
+1. First, you'll review a barebones Codec Server implementation in Go, and make
+   the necessary changes to integrate the custom data converter from Exercise 1.
+   Examine the `main.go` file in the `codec-server` subdirectory. This file
+   contains a complete HTTP server implementation using the Go standard library.
+   It listens on endpoints at `/{namespace}/encode` and `/{namespace}/decode` as
+   expected by the Temporal CLI, Web UI, and SDKs, and contains stubs for OIDC
+   and CORS enablement. These are the baseline requirements for a Temporal Codec
+   Server, which can be implemented using standard HTTP functionality in any
+   language of your choosing.
+2. Temporal Codec Servers need, at minimum, one additional configuration detail
+   before they can be deployed from sample code. Specifically, Codec Servers
+   need to import the Converter logic from your own application, and then map
+   the Converter logic on a per-Namespace basis. Edit the `import()` block at
+   the top of `codec-server/main.go` to import the rest of your application as a
+   Go module named `temporalconverters`. Don't forget that you can import Go
+   modules from GitHub URLs.
+3. Next, create an array named `codecs` in the `main()` function of
+   `codec-server/main.go`. Keys should be Namespace strings. Values should be of
+   the type `converter.PayloadCodec`. By default, you only need to assign the
+   `default` namespace to `{temporalconverters.NewPayloadCodec()}` from this
+   example.
+4. After making these additions, you should have a functioning Codec Server,
+   integrated with your application logic. Again, everything else in here is
+   configured as generically as possible — note that this example Codec Server
+   listens on port 8081, which is usually used in testing configurations — but
+   this fulfills all the requirements of a Temporal Codec Server, and you could
+   incorporate any other authentication requirements on top of HTTP as needed.
+   Run your Codec Server with `go run ./codec-server` from the root of your
+   project directory. This will block the terminal it runs in, and await
+   connections.
+5. Now you can retrieve the decoded output of your Workflow Execution from the
+   previous Exercise. From another terminal window, run `temporal workflow show \
+   -w converters_workflowID --codec-endpoint 'http://localhost:8081/{namespace}'`.
+   It should retain the same Event History as before, with the decoded result
```

Comment:
Since the Workflow Execution run at the end of the previous exercise intentionally throws an error, in order to test the Failure Converter, this does not work as you describe. Instead, I get this:

```
Progress:
  ID          Time                    Type
   1  2024-04-03T21:13:14Z  WorkflowExecutionStarted
   2  2024-04-03T21:13:14Z  WorkflowTaskScheduled
   3  2024-04-03T21:13:14Z  WorkflowTaskStarted
   4  2024-04-03T21:13:14Z  WorkflowTaskCompleted
   5  2024-04-03T21:13:14Z  ActivityTaskScheduled
   6  2024-04-03T21:13:14Z  ActivityTaskStarted
   7  2024-04-03T21:13:14Z  ActivityTaskCompleted
   8  2024-04-03T21:13:14Z  WorkflowTaskScheduled
   9  2024-04-03T21:13:14Z  WorkflowTaskStarted
  10  2024-04-03T21:13:14Z  WorkflowTaskCompleted
  11  2024-04-03T21:13:14Z  WorkflowExecutionFailed

Result:
  Status: FAILED
  Failure: &Failure{Message:Encoded failure,Source:GoSDK,StackTrace:,Cause:nil,FailureType:Failure_ApplicationFailureInfo,}
```

I think you need to have steps preceding this one that asks the learner to run the Worker and starter in this exercise, since that will generate a new Workflow Execution that runs to completion.

Reply:
Oh! Very good catch! Fixed.